orc8r/cloud/go/services/ctraced/obsidian/handlers/handlers.go

```diff
@@ -0,0 +1,246 @@
+		if err != nil {
+			return err
+		}
+		if resp.Success == false {
+			return obsidian.HttpError(errors.New("Failed to end call trace"), http.StatusInternalServerError)
+		}
+
+		err = storage.StoreCallTrace(networkID, callTraceID, resp.TraceContent)
+		if err != nil {
+			return obsidian.HttpError(errors.Wrap(err, "failed to save call trace data"), http.StatusInternalServerError)
+		}
+
+		_, err = configurator.UpdateEntity(networkID, mutableCallTrace.ToEntityUpdateCriteria(callTraceID, *callTrace), serdes.Entity)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		return c.NoContent(http.StatusNoContent)
+	}
+}
+
+func GetDeleteCallTraceHandlerFunc(client GWCtracedClient, storage storage.CtracedStorage) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		networkID, callTraceID, nerr := getNetworkIDAndCallTraceID(c)
+		if nerr != nil {
+			return nerr
+		}
+
+		err := storage.DeleteCallTrace(networkID, callTraceID)
+		if err != nil {
+			return obsidian.HttpError(errors.Wrap(err, "failed to delete call trace data"), http.StatusInternalServerError)
+		}
+
+		err = configurator.DeleteEntity(networkID, orc8r.CallTraceEntityType, callTraceID)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		return c.NoContent(http.StatusNoContent)
+	}
+}
+
+func GetDownloadCallTraceHandlerFunc(storage storage.CtracedStorage) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		networkID, callTraceID, nerr := getNetworkIDAndCallTraceID(c)
+		if nerr != nil {
+			return nerr
+		}
+
+		callTrace, err := storage.GetCallTrace(networkID, callTraceID)
+		if err != nil {
+			return obsidian.HttpError(errors.Wrap(err, "failed to retrieve call trace data"), http.StatusInternalServerError)
+		}
+
+		return c.Blob(http.StatusOK, "application/pcap", callTrace)
+	}
+}
+
+func getCallTrace(c echo.Context) (*models.CallTrace, error) {
+	networkID, callTraceID, nerr := getNetworkIDAndCallTraceID(c)
+	if nerr != nil {
+		return nil, nerr
+	}
+	ent, err := configurator.LoadEntity(
+		networkID, orc8r.CallTraceEntityType, callTraceID,
+		configurator.EntityLoadCriteria{LoadConfig: true},
+		serdes.Entity,
+	)
+	if err == merrors.ErrNotFound {
+		return nil, obsidian.HttpError(err, http.StatusNotFound)
+	}
+	if err != nil {
+		return nil, obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	callTrace := &models.CallTrace{}
+	err = callTrace.FromBackendModels(ent)
+	if err != nil {
+		return nil, obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	return callTrace, nil
+}
+
+func getNetworkIDAndCallTraceID(c echo.Context) (string, string, *echo.HTTPError) {
+	vals, err := obsidian.GetParamValues(c, pathParamNetworkID, pathParamTraceID)
+	if err != nil {
+		return "", "", err
+	}
+	return vals[0], vals[1], nil
+}
+
+func buildStartTraceRequest(cfg *models.CallTraceConfig) (*protos.StartTraceRequest, error) {
```

Comment:
minor nit: it doesn't need to return error. function seems to not do much, are we missing something here? Seems like we can rather move this to the caller. Is this for future usecase ?

Reply:
For future usecase